samples/kmeans-clustering/Inference.ipynb

```diff
@@ -0,0 +1,181 @@
+    "        return None\n",
+    "\n",
+    "    def predict(self, X: np.ndarray, feature_names: Iterable[str], meta: Dict = None) -> Union[np.ndarray, List, str, bytes]:\n",
+    "        \"\"\"Takes an array (numpy) X and feature_names and returns an array of predictions.\n",
+    "\n",
+    "        Args:\n",
+    "            X (numpy.array): Array-like with data.\n",
+    "            feature_names (iterable, optional): Array of feature names.\n",
+    "            meta (dict, optional): Dict of metadata.\n",
+    "        \"\"\"\n",
+    "        # Put data in a pandas.DataFrame to sort it\n",
+    "        df = pd.DataFrame(X, columns=feature_names)\n",
+    "\n",
+    "        # Put data back in a numpy.ndarray\n",
+    "        X = df[self.columns].to_numpy()\n",
+    "\n",
+    "        y_pred = self.pipeline.predict(X)\n",
+    "\n",
+    "        return y_pred"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## API Contract\n",
+    "\n",
+    "There are two sections:\n",
+    "\n",
+    "- `features` : The feature array you intend to send in a request\n",
+    "\n",
+    "Each section has a list of definitions. Each definition consists of:\n",
+    "\n",
+    "- `name` : String : The name of the feature\n",
+    "- `ftype` : one of CONTINUOUS, CATEGORICAL : the type of the feature\n",
+    "- `dtype` : One of FLOAT, INT : Required for ftype CONTINUOUS : What type of feature to create\n",
+    "- `values` : list of Strings : Required for ftype CATEGORICAL : The possible categorical values\n",
+    "- `range` : list of two numbers : Optional for ftype CONTINUOUS : The range of values (inclusive) that a continuous value can take\n",
+    "- `repeat` : integer : Optional value for how many times to repeat this value\n",
+    "- `shape` : array of integers : Optional value for the shape of array to coerce the values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile contract.json\n",
+    "{\n",
+    "    \"features\": [\n",
+    "        {\n",
+    "            \"name\": \"SepalLengthCm\",\n",
+    "            \"dtype\": \"FLOAT\",\n",
+    "            \"ftype\": \"continuous\",\n",
+    "            \"range\": [0.0, 7.0]\n",
+    "        },\n",
+    "        {\n",
+    "            \"name\": \"SepalWidthCm\",\n",
+    "            \"dtype\": \"FLOAT\",\n",
+    "            \"ftype\": \"continuous\",\n",
+    "            \"range\": [0.0, 4.0]\n",
+    "        },\n",
+    "        {\n",
+    "            \"name\": \"PetalLengthCm\",\n",
+    "            \"dtype\": \"FLOAT\",\n",
+    "            \"ftype\": \"continuous\",\n",
+    "            \"range\": [0.0, 7.0]\n",
+    "        },\n",
+    "        {\n",
+    "            \"name\": \"PetalWidthCm\",\n",
+    "            \"dtype\": \"FLOAT\",\n",
+    "            \"ftype\": \"continuous\",\n",
+    "            \"range\": [0.1, 3.0]\n",
```

Comment:
O contract.json precisa de um `target`, que são as colunas que você retorna no predict (y_pred).
Se for só um `cluster` poderia ser:
```
%%writefile contract.json
{
    "features": [
        {
            "name": "SepalLengthCm",
            "dtype": "FLOAT",
            "ftype": "continuous",
            "range": [0.0, 7.0]
        },
        {
            "name": "SepalWidthCm",
            "dtype": "FLOAT",
            "ftype": "continuous",
            "range": [0.0, 4.0]
        },
        {
            "name": "PetalLengthCm",
            "dtype": "FLOAT",
            "ftype": "continuous",
            "range": [0.0, 7.0]
        },
        {
            "name": "PetalWidthCm",
            "dtype": "FLOAT",
            "ftype": "continuous",
            "range": [0.1, 3.0]
        }
    ],
    "targets": [
        {
            "name": "cluster",
            "ftype": "categorical",
            "dtype": "INT",
            "values": [1, 2, 3]
        }
    ]
}
```